Philips.CodeAnalysis.Test/Maintainability/Maintainability/ReduceCognitiveLoadAnalyzerTest.cs

```diff
@@ -0,0 +1,469 @@
+		}
+
+		[TestMethod]
+		public void CognitiveLoadBang()
+		{
+			const string template = @"
+class Foo
+{
+	private void Test()
+	{
+		if (!(1==1)) {}
+	}
+}
+";
+			VerifyCSharpDiagnostic(template, DiagnosticResultHelper.Create(DiagnosticIds.ReduceCognitiveLoad, new Regex("Load of 3 ")));
+		}
+
+		[TestMethod]
+		public void CognitiveLoadNested1()
+		{
+			const string template = @"
+class Foo
+{
+	private void Test()
+	{
+		if (1==1)
+		{
+			// Nested block statement counts double
+			if (2 == 2) {}
+		}
+	}
+}
+";
+			VerifyCSharpDiagnostic(template, DiagnosticResultHelper.Create(DiagnosticIds.ReduceCognitiveLoad, new Regex("Load of 4 ")));
+		}
+
+		[TestMethod]
+		public void CognitiveLoadNested2()
+		{
+			const string template = @"
+class Foo
+{
+	private void Test()
+	{
+		if (1==1)
+		{
+			// Nested block statement counts double
+			if (2 == 2) {}
+			// Nested block statement counts double
+			if (3 == 3) {}
+		}
+	}
+}
+";
+			VerifyCSharpDiagnostic(template, DiagnosticResultHelper.Create(DiagnosticIds.ReduceCognitiveLoad, new Regex("Load of 6 ")));
+		}
+
+		[TestMethod]
+		public void CognitiveLoadNested3()
+		{
+			const string template = @"
+class Foo
+{
+	private void Test()
+	{
+		if (1==1)
+		{
+			// Nested block statement counts double
+			if (2 == 2)
+			{
+				// Nested block statement counts more than triple - it's exponential
+				if (3 == 3) {}
+			}
+		}
+	}
+}
+";
+			VerifyCSharpDiagnostic(template, DiagnosticResultHelper.Create(DiagnosticIds.ReduceCognitiveLoad, new Regex("Load of 8 ")));
+		}
+
+
+
+		[TestMethod]
+		public void CognitiveLoadCombo()
+		{
+			const string template = @"
+class Foo
+{
+	private void Test()
+	{ //1
+		if(1==1)
+		{ //2
+			if(2==2) {return;} //3,4,5
+		}
+		if(3==3)
+		{ return;} // 6,7
+	}
+}
+";
+			VerifyCSharpDiagnostic(template, DiagnosticResultHelper.Create(DiagnosticIds.ReduceCognitiveLoad, new Regex("Load of 7 ")));
+		}
+
+		[TestMethod]
+		public void CognitiveLoad3()
+		{
+			const string template = @"
+class Foo
+{
+	private void Test()
+	{ //1
+		int dynamicDataCount = 0;
+		int dataRowCount = 0;
+		bool hasTestSource = false;
+		foreach (AttributeSyntax attribute in methodDeclaration.AttributeLists.SelectMany(x => x.Attributes))
+		{ //2
+			if (Helper.IsDataRowAttribute(attribute, context))
+			{ //3,4
+				dataRowCount++;
+				continue; //5
+			}
+			if (Helper.IsAttribute(attribute, context, MsTestFrameworkDefinitions.DynamicDataAttribute, out _, out _))
+			{ //6,7
+				dynamicDataCount++;
+				continue; //8
+			}
+			SymbolInfo symbol = context.SemanticModel.GetSymbolInfo(attribute);
+			if (symbol.Symbol is IMethodSymbol method)
+			{ //9,10
+				if (method.ContainingType.AllInterfaces.Contains(Definitions.ITestSourceSymbol))
+				{ //11,12,13,14
+					hasTestSource = true;
+				}
+			}
+		}
+		if (isDataTestMethod)
+		{ //15
+			if (dataRowCount != 0 && dynamicDataCount == 0) //16,17
+			{ // 18,19
```

Comment:
You add 2 for this block, Do you also count the absent `else` here ?

Reply:
This gets a double hit because it's nested. (Had there been an `else` block, it would've been yet another double hit.